### PR TITLE
Closes #1336: A utility that periodically checks if storage system contains files that it should according to what we have in Dataverse database

### LINKED ISSUE #1336: A utility that periodically checks if storage system contains files that it should according to what we have in Dataverse database
1. It should be possible to set the check period.
2. Main Dataverse e-mail receives a message that a check has been performed and that some discrepancies were found (if there were any).
3. A Superuser should be able to start an additional check on demand via API.

### PR DESCRIPTION
**Pull request by tomaszblazejczyk**

### CODE REVIEW

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/FileService.java [hunk 1]
```diff
@@ -204,4 +216,46 @@ public String scan(InputStream fileInput) throws IOException {
 
     }
 
+    public String checkFilesIntegrity() {
```
Comment (madryk) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/FileService.java]: IMO dedicated model class as result type would be much better. Not much can be done further with string.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/FileService.java [hunk 2]
```diff
@@ -204,4 +216,46 @@ public String scan(InputStream fileInput) throws IOException {
...
+            messageBodyBuffer.append("Number of files with failures: " + suspicious.size() + "\n\n");
+            messageBodyBuffer.append("List of files with failures:\n");
+            suspicious.stream().forEach(datafile -> messageBodyBuffer.append("File id: " + datafile.getId() + ", file label: " + datafile.getLatestFileMetadata().getLabel() + "\n"));
+            
+            String messageSubject = "Dataverse files integrity check report";
+            
+            authSvc.findAllAuthenticatedUsers().
+            stream().
+            filter(user -> user.isSuperuser()).
+            forEach(user -> mailService.sendMailAsync(user.getEmail(), new EmailContent(messageSubject, messageBodyBuffer.toString(), "")));
```
Comment (madryk) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/FileService.java]: indentation issue

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/FileService.java [hunk 3]
```diff
@@ -204,4 +216,46 @@ public String scan(InputStream fileInput) throws IOException {
...
+        if (dataFiles.isEmpty()) {
+            return "No datafiles found - check database.";
+        } else {
+
+            int filesCount = dataFiles.size();
+            String message = "Found " + filesCount + " files in repository.";
+            for (DataFile dataFile:dataFiles) {
+                try {
+                    StorageIO<DataFile> storageIO = DataAccess.dataAccess().getStorageIO(dataFile);
+                    if (!storageIO.exists() && storageIO.getSize() == dataFile.getFilesize()) {
```
Comment (madryk) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/FileService.java]: Can you point how `storageIO.getSize()` is populated. I can only find reference to setting this value when `open()` is executed. In other cases seems to me like it's always 0

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/Admin.java [hunk 4]
```diff
@@ -60,43 +100,6 @@
...
-import java.util.Date;
-import java.util.List;
-import java.util.Map;
-import java.util.logging.Level;
-import java.util.logging.Logger;
-
-import static edu.harvard.iq.dataverse.common.NullSafeJsonBuilder.jsonObjectBuilder;
-import static edu.harvard.iq.dataverse.util.json.JsonPrinter.json;
-import static edu.harvard.iq.dataverse.util.json.JsonPrinter.rolesToJson;
-import static edu.harvard.iq.dataverse.util.json.JsonPrinter.toJsonArray;
```
Comment (madryk) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/Admin.java]: Can you set your ide to organize imports: first group: all non static, next group: javax.* and java.*, last group: static imports? If we continue to mixing up organize order then we will have a lot of changes only for imports organizing

**Review by madryk — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/FileService.java (hunk 3 above)
Comment (tomaszblazejczyk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/FileService.java]: I admit I was a bit messy - now should work properly

**Review by madryk — APPROVED** [review]

**Pull request merged**